javascript/react/common-mistakes-in-react/setstate-is-async.md

```diff
@@ -0,0 +1,100 @@
+---
+author: kapnobatai136
+  
+aspects:
+
+  - workout
+
+  - deep
+
+type: normal
+
+category: tip
+
+---
+
+# setState is async
+
+---
+## Content
+
+More often than not, we are very eager with accessing a new state value after setting it. Because a new value is set on the next available render, the state might not reflect your latest update. Let's take a look at an example:
```

Comment:
```suggestion
More often than not, we are eager with accessing a new state value after setting it. Because a new value is set on the next available render, the state might not reflect your latest update. Let's take a look at an example:
```